### DISCUSSION TITLE
rtsp video stream stable? or is it just me

### DISCUSSION THREAD

**Discussion by vincom** [gtxaspec/wz_mini_hacks#267] [Q&A] [answered]
i tried the wyze rtsp fw, but getting the stream freezes or it loses stream, this is using blue iris v5.
thought i try mini hack to get usb ethernet to see if its more stable, its not. everything is working as it should but losing the video feed is annoying. luckily i got a few of these cameras for free, because if i had to buy the cam, sdcard and an ethernet adpapter i would rather just buy a sub $100can onvif  ip camera from amazon. i got a couple of older cheap ip cams without these issues.
so is this a common issue with v3 cam rtsp stream or is it just me

**Comment by gtxaspec** [comment]
try using udp mode for whatever client you are using, or, if already using udp, try tcp and see if that's better.

**Reply by vincom** [reply]
> try using udp mode for whatever client you are using, or, if already using udp, try tcp and see if that's better.

 so u r saying u dont? what client are u using to view ur stream. are u using an nvr?
im using blueiris v5, not sure how i would set it to use udp or tcp correctly in bi.
everyone on the wyze rtsp thread have been having stabilty issues with the stream, either with wyzes rtsp fw or mini hacks

**Comment by gtxaspec** [comment]
using tinycampro and vlc running a mosaic, no issues here.

**Reply by vincom** [reply]
> using tinycampro and vlc running a mosaic, no issues here.

so ur not using an nvr to record? eg blueiris

**Comment by guarddog13** [comment]
I use AgentDVR but don't record. There's A LOT of framerate drops. I actually blame AgentDVR. I put the streams on Onvier on my Shield Pro and can play 6 cameras in multi-view...once I add a 7th I think I'm overloading as the 7th will load but freeze. I can view all 7 in TinyCam (non-Pro), on my phone, and am considering putting TinyCam Pro on the Shield Pro along with Tasker (as I see Tinycam has a tasker plugin), using tinycam and tasker I can call a HA service to Deepstack and maybe skip AgentDVR. I'm trying to use the cameras as people motion detectors with deepstack. My laptop can handle 7 cameras at 40%, 60 when the house is busy, but the Nvidia Shield is fairly powerful and I could free my laptop back up.

**Comment by vincom** [comment]
w/ blue iris it loses the stream for up to a couple of minutes at a time. i tried varied settings in blue iris but nothing is consistent, random connection drop. that my reason for the rtsp hack is to get a nvr to record reliably. b4 i was blaming wyze's rtsp fw but it could be blue iris. im going to try other nvr's to see if i can find a compatible one.

on a side note, users at the wyze rtsp forum have stated that if the camera cant communicate with its wyzr servers it will stop streaming for x amount of minutes at a time, ive not tested this yet, i will, just need my ca,era to be wan capable at the moment, but i was going to drop its internet connection via my router and see what happens

**Comment by guarddog13** [comment]
I don't see any actual stream drops and deepstack is fairly reliable. My main issue is the artifacting. It works fine on my Android phone and two different apps on my shield pro. Hypothetically the issue either falls with my laptop's decoding or with AgentDVR. It seems to get a good stream on vlc on my laptop so that points to AgentDVR.

**Comment by vincom** [comment]
with the right settings and config blueiris doesnt have any real artifacts, i would say not any but ymmv, but it does lose the stream randomly. since youre not having stream connection issues then it must be a setting within bi, ill keep tinkering.

did u notice if cam is not able to get online but is active on lan that you will lose the stream for minutes at a time?

**Reply by guarddog13** [reply]
So I have found that the low res streams are much much more stable. I have zero frame rate drops and zero disconnects using the low res stream on AgentDVR. Using this on GPU with Deepstack on my GPU as well, I have nearly perfect person detection. These are my internal cameras and I just use the SD card to record. This utilizes only 20% of CPU and about 20% of the GPU...with 7 cameras.

I do use an external camera to grab 5 second clips to send to my phone and having the high res link in the recording section for some reason it's not loading the high res feed and only recording the low res feed. They also have a setting when you maximum the camera it's supposed to switch to the high res feed. For some reason it just won't load it. If I set the high res feed as the main feed it works fine...although this means I get the frame drops and artifacts again. The 5 sec clip was more a novetly anyway as I have a nest doorbell in the same area that grabs gifs to send as a notification. I may put wz_mini on this camera again to see if it's just a corrupted load.

**Comment by guarddog13** [comment]
I havent had any issues getting any of them to stay online so couldn't say. 

Currently I'm trying to fine tune my curl posts to AgentDVR/deepstack..its about 95-99% reliable based on how busy the house is and time of day (lights that shine into the house)...

**Comment by vincom** [comment, marked as answer]
i think it was misconfigured blue iris settings, all things seem good now, even set a camera with no internet and no loss of stream, all setup for hi res 2mb
its been about 24 hrs with the new blueiris v5 settings.  it all seems good. testing 3 separate cameras at the same time, all  v3.
each camera has 2 profiles in blue iris, timelapse record and motion detection to trigger record w/no time lapse.
**make sure both  profiles for each cam match video source configuration, bi doesnt behave correctly if same ip camera is configured differently**.

first camera: on stock fw w/mini hack w/ethernet (better kb rate than wifi)
second camera:  on stock fw w/mini hack wifi
third camera: wyze rtsp fw, no mini hack(last 12hrs no internet access)

bi set to rtsp as ip video source
setup to record every 15min interval, time lapse 3fps, record lenght was 2m15s/
every hour of record i might lose a few seconds due to lose of stream, once inawhile ~1min. so for a 24hr period maybe 2min lost, ymmv
it didnt not matter if camera did not have internet access, wyze app could not connect to it, it did not have  any effect in streaming video even with the wyze rtsp fw. ymmv

blueiris config pics shown here, i somehow cant get pics to link to this post correctly
https://forums.wyzecam.com/t/beta-testing-for-wyze-cam-v3-rtsp-firmware-now-available/196922/754?u=vincom

**Reply by guarddog13** [reply]
I wish I could get my cameras on ethernet. I don't have the time or patience to run that many cables to my switches or the router nodes. I have 1, maybe 2 cameras, of my 7 that are close enough to run to a network switch and they aren't really all that close.

Do you have any luck getting the mics to work on Bi? They don't work on AgentDVR. I can't cut off access to wyze app without the mic. I can use tinycampro for the mic but that's connecting P2P and I need my wyze creds. Using the wz_mini links in tinycampro the mic doesn't work either.

**Reply by vincom** [reply]
> I wish I could get my cameras on ethernet. I don't have the time or patience to run that many cables to my switches or the router nodes. I have 1, maybe 2 cameras, of my 7 that are close enough to run to a network switch and they aren't really all that close.
> 
> Do you have any luck getting the mics to work on Bi? They don't work on AgentDVR. I can't cut off access to wyze app without the mic. I can use tinycampro for the mic but that's connecting P2P and I need my wyze creds. Using the wz_mini links in tinycampro the mic doesn't work either.

i thought i could solve my loosing stream problem  w/wired ethernet, but after configuring bi  the wifi seems very stable on three other cameras as compared to one on ethernet. 2 of them are about 25ft away from my asus ac86u router, so i wouldnt fret too much about trying to run ethernet canbles, the most i got was a slightly better bit rate.
i have not set up any audio for the cams...but i can try and report back on the mics

edit. i cant , i have blue iris running on a server w/no audio hardware

**Comment by teredactle** [comment]
> i tried the wyze rtsp fw, but getting the stream freezes or it loses stream, this is using blue iris v5. thought i try mini hack to get usb ethernet to see if its more stable, its not. everything is working as it should but losing the video feed is annoying. luckily i got a few of these cameras for free, because if i had to buy the cam, sdcard and an ethernet adpapter i would rather just buy a sub $100can onvif ip camera from amazon. i got a couple of older cheap ip cams without these issues. so is this a common issue with v3 cam rtsp stream or is it just me

How is the audio? I've noticed that both on the v2 and v3, audio is total garbage.
If you pull the files off the SD card, the audio is fine.

**Reply by vincom** [reply]
im not using audio . but to address my previous post on ethernet reliablity, ethernet is actually stable, streaming and/or frames are not dropping out,  as far as i can tell

**Reply by teredactle** [reply]
> im not using audio . but to address my previous post on ethernet reliablity, ethernet is actually stable, streaming and/or frames are not dropping out, as far as i can tell

Are you using defaults for settings FPS, bitrate, etc? Or have you changed some things. I.e. the WC3 is 20fps day, 15fps night, can you make it do 25fps both night and day? Is it capable? Also can you boost the bitrate? If you configure it this way for the mini hack, does that mean it would also change those setting when looking at it through the wyze app? Using the POE dongle I'd be looking at boosting some of those settings...

**Reply by vincom** [reply]
RTSP_HI_RES_FPS
20
RTSP_HI_RES_MAX_BITRATE
2048
RTSP_HI_RES_TARGET_BITRATE
1024

wyze app only has sd or hd as option, i dont record via wyze app
i got  the camera to be able to use it as a stand alone device and not be dependent on the app
